content/kubeone/master/guides/operating-system-manager/usage/_index.en.md

```diff
@@ -0,0 +1,145 @@
++++
+title = "Working with Operating System Manager"
+date = 2022-08-05T12:00:00+02:00
+enableToc = true
++++
+
+## Enable/Disable OSM
+
+Starting with KubeOne 1.5, OSM will be enabled by default for new and existing clusters. This change will be applied to existing clusters when they upgrade to 1.5.
+
+To fallback to legacy user-data from Machine Controller, we can disable OSM for a KubeOne cluster. Although this is not recommended for production environments.
+
+```yaml
+apiVersion: kubeone.k8c.io/v1beta2
+kind: KubeOneCluster
+versions:
+  kubernetes: 1.23.6
+addons:
+  enable: true
+operatingSystemManager:
+  deploy: false
+```
+
+## Migrating existing machine deployments
+
+{{% notice warning %}}
+Existing Machine Deployments will not be rotated automatically. Existing machine deployment
+should be rotated manually, for them to consume configurations generated by OSM.
+{{% /notice %}}
+
+To perform this rotation please follow the guide at [Rolling Restart MachineDeploments][rolling-restart].
+
+[rolling-restart]: {{< ref "../../../cheat_sheets/rollout_machinedeployment" >}}
+
+## Using custom OperatingSystemProfile
+
+The annotation `k8c.io/operating-system-profile` is used at the Machine Deployment level to select the Operating System Profile that will be used to generate configurations for provisioning the machines.
+
+### When creating a new cluster
+
+For new clusters, a dedicated terraform variable `initial_machinedeployment_operating_system_profile` was introduced for this purpose. It'll propagate the specified value to the initial machine deployments in the cluster.
+
+It can be used as follows:
+
+1. Create a custom OSP and place it in a folder inside your custom addons path, if it exists otherwise create one. Let's take this minimalistic OSP just for the sake of understanding:
```

Comment:
```suggestion
1. Create a custom OSP and place it in a folder inside your custom addons path, if it exists, otherwise create one. Let's take this minimalistic OSP as an example:
```